### PR TITLE
[Fix] Group Task Retry including nested tasks does not update parent id of copied tasks.

### PR DESCRIPTION
**Pull request by nak-tera**
https://github.com/treasure-data/digdag/pull/738 added dependencies for copied tasks but didn’t do parent id.
So we add the implements that update parent id of copied tasks.
 
Without updating parent id, Group Retry unexpectedly works as follow.
Thanks.

# issue
Workflow Definition
group_retry.dig
```
+wf:
  _retry: 3

  +t1:
    echo>: foo

  +t2:
    +t2_nested:
      echo>: bar

  +t3:
    fail>: failure
```
Result

We expect task 384 has task 383 as a parent, but actually, the parent id is 379.
So task 384 belongs to task group that has task 379 as a parent.
As a result, task 384 and task 382 start processing parallelly.

# After Fixed

### CODE REVIEW

File: digdag-core/src/main/java/io/digdag/core/database/DatabaseSessionStoreManager.java [hunk 1]
```diff
@@ -886,6 +886,7 @@ public boolean copyInitialTasksForRetry(List<Long> recursiveChildrenIdList)
             for (StoredTask task : tasks) {
                 Task newTask = Task.taskBuilder()
                     .from(task)
+                    .parentId(task.getParentId().transform(it -> oldIdToNewId.containsKey(it) ? oldIdToNewId.get(it) : it))
```
Comment (komamitsu) [inline digdag-core/src/main/java/io/digdag/core/database/DatabaseSessionStoreManager.java]: How about simply `it -> oldIdToNewId.getOrDefault(it, it)`?

**Comment by komamitsu** [comment]
Left a minor comment. Overall LGTM!

**Comment by muga** [comment]
@nak-tera Thank you. LGTM too. Can you help to add tests to confirm if this correctly works or not?

File: digdag-core/src/main/java/io/digdag/core/database/DatabaseSessionStoreManager.java (hunk 1 above)
Comment (nak-tera) [reply digdag-core/src/main/java/io/digdag/core/database/DatabaseSessionStoreManager.java]: It's better! I'll change it.

**Comment by nak-tera** [comment]
@muga OK, I'll do it this week.

**Comment by nak-tera** [comment]
@muga I added the tests. Can you check them?

File: digdag-core/src/test/java/io/digdag/core/workflow/TaskDependenciesTest.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    }
+
+    @After
+    public void destroy()
+            throws Exception {
+        embed.close();
+    }
+
+    @Test
+    public void TaskDependenciesTest()
```
Comment (muga) [inline digdag-core/src/test/java/io/digdag/core/workflow/TaskDependenciesTest.java]: Do you have any reasons why first letter of the method name is not lowercase? How about `testTaskDependencies`?

File: digdag-core/src/test/java/io/digdag/core/workflow/TaskDependenciesTest.java [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+            assertThat(tasks.get(2).getUpstreams(), is(Collections.singletonList(tasks.get(1).getId())));
+            assertThat(tasks.get(3).getUpstreams(), is(Collections.emptyList()));
+            assertThat(tasks.get(4).getUpstreams(), is(Collections.emptyList()));
+
+            return null;
+        });
+    }
+
+    @Test
+    public void GroupRetryTaskDependenciesTest()
```
Comment (muga) [inline digdag-core/src/test/java/io/digdag/core/workflow/TaskDependenciesTest.java]: ditto

File: digdag-core/src/test/java/io/digdag/core/workflow/TaskDependenciesTest.java (hunk 2 above)
Comment (nak-tera) [reply digdag-core/src/test/java/io/digdag/core/workflow/TaskDependenciesTest.java]: Oh, I made a mistake...
I wil fix it.

**Comment by nak-tera** [comment]
@muga Sorry for the late reply.
Thank you for your comments. I fixed it.

**Comment by muga** [comment]
@nak-tera thank you! merging.. @komamitsu

**Pull request merged**